tests/__snapshots__/test_application-test_client_side_reports

```diff
@@ -941,6 +941,43 @@
   "links": [
     "test-link"
   ],
+  "narrations": [
```

Comment:
This part is problematic. In my case, all these narrations would e.g. double the size of the balance sheet report. I think we should switch to lazily loading them only when the dialog is used.

Reply:
okay yeah, I can understand the problem. Can you give me a hint for a possible implementation though? Or do you have something similar inside the repo where I could get some inspiration? 

I just looked at it and with my current understanding I would have to remove the narrations from the `get_ledger_data()` method inside `internal_api.py` and instead create another endpoint in `json_api.py` which is then somehow called once the dialog is opened? Is this remotely a good way to solve this or is there maybe an easier way?